src/manage_sql_report_formats.c

```diff
@@ -2533,6 +2545,86 @@ report_format_validate_param_value (report_format_t report_format,
             return 0;
         }
         break;
+      case REPORT_FORMAT_PARAM_TYPE_MULTI_SELECTION:
+        {
+          long long int min, max, actual;
+          min = report_format_param_type_min (report_format, name);
+          max = report_format_param_type_max (report_format, name);
+          actual = 0LL;
+          cJSON *json = cJSON_Parse (value);
+          cJSON *array_item = NULL;
+
+          if (!cJSON_IsArray (json))
+            {
+              if (error_message)
+                {
+                  *error_message
+                    = g_strdup_printf ("value of param \"%s\""
+                                       " is not a valid JSON array",
+                                       name);
+                }
+              cJSON_Delete (json);
+              return 1;
+            }
+
+          cJSON_ArrayForEach (array_item, json)
+            {
+              char *string;
+              if (!cJSON_IsString (array_item))
+                {
+                  if (error_message)
+                    {
+                      *error_message
+                        = g_strdup_printf ("value of param \"%s\""
+                                           " contains a non-string value",
+                                           name);
+                    }
+                  cJSON_Delete (json);
```

Comment:
I think I would be cautious about deleting the array while iterating over it. I would prefer to either set a flag and break or jump out of the loop and delete the array afterwards.

Reply:
The function returns immediately after this so this should be fine.